src/ui/MLFacetManager/MLFacetManager.ts

```diff
@@ -0,0 +1,225 @@
+import { Component } from '../Base/Component';
+import { MLFacet } from '../MLFacet/MLFacet';
+import { InitializationEvents } from '../../events/InitializationEvents';
+import { QueryEvents, IQuerySuccessEventArgs } from '../../events/QueryEvents';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { exportGlobally } from '../../GlobalExports';
+import { find } from 'underscore';
+import { IFacetResponse } from '../../rest/Facet/FacetResponse';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { ComponentOptions } from '../Base/ComponentOptions';
+
+export interface IMLFacetManagerOptions {
+  enableReorder?: boolean;
+  onUpdate?: IMLFacetManagerOnUpdate;
+  compareFacets?: IMLFacetManagerCompareFacet;
+}
+
+export interface IMLFacetManagerOnUpdate {
+  (facet: MLFacet, index: number): void;
+}
+
+export interface IMLFacetManagerCompareFacet {
+  (facetA: MLFacet, facetB: MLFacet): number;
+}
+
+export class MLFacetManager extends Component {
+  static ID = 'MLFacetManager';
+  static doExport = () => exportGlobally({ MLFacetManager });
+
+  /**
+   * The options for the MLFacet
+   * @componentOptions
+   */
+  static options: IMLFacetManagerOptions = {
+    /**
+     * Whether to allow the reordering of the facets depending on the server's response.
```

Comment:
of the facets -> of facets
depending on the server's response. -> based on Coveo ML and index ranking scores.